From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Fri, 22 Jan 2016 23:21:10 +0800
Subject: crypto: skcipher - Add helper to zero stack request
Git-commit: 1aaa753d918c48c603195a468766e6a2b32b87f9
Patch-mainline: v4.6-rc1
References: FATE#322288

As the size of an skcipher_request is variable, it's awkward to
zero it explicitly.  This patch adds a helper to do that which
should be used when it is created on the stack.

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 include/crypto/skcipher.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/crypto/skcipher.h b/include/crypto/skcipher.h
index 2f07b4fce3e0..f7a03a17bc1d 100644
--- a/include/crypto/skcipher.h
+++ b/include/crypto/skcipher.h
@@ -445,6 +445,13 @@ static inline void skcipher_request_free(struct skcipher_request *req)
 	kzfree(req);
 }
 
+static inline void skcipher_request_zero(struct skcipher_request *req)
+{
+	struct crypto_skcipher *tfm = crypto_skcipher_reqtfm(req);
+
+	memzero_explicit(req, sizeof(*req) + crypto_skcipher_reqsize(tfm));
+}
+
 /**
  * skcipher_request_set_callback() - set asynchronous callback function
  * @req: request handle

